integration_tests/src/main/python/asserts.py

```diff
@@ -289,8 +289,9 @@ def assert_gpu_and_cpu_sql_writes_are_equal_collect(table_name_factory, write_sq
     cpu_table = table_name_factory.get()
     cpu_start = time.time()
     def do_write(spark, table_name):
-        sql_text = write_sql_func(spark, table_name)
-        spark.sql(sql_text)
+        sql_text_array = write_sql_func(spark, table_name)
+        for sql_text in sql_text_array:
+            spark.sql(sql_text)
```

Comment:
Ideally this should auto-detect whether there's multiple lines here or not rather than force all existing clients to change.
```suggestion
        sql_texts = write_sql_func(spark, table_name)
        sql_text_list = [sql_texts] if isinstance(sql_texts, str) else sql_texts
        for sql_text in sql_text_list:
            spark.sql(sql_text)
```

Reply:
Nice one.